tests/manage/nfs_feature/test_nfs_feature_enable_for_ODF_clusters.py

```diff
@@ -0,0 +1,449 @@
+import pytest
+import logging
+import yaml
+
+
+from ocs_ci.ocs import api_client, constants, ocp
+from ocs_ci.helpers import helpers
+from functools import partial
+from ocs_ci.framework.testlib import (
+    skipif_ocs_version,
+    ManageTest,
+    tier1,
+    skipif_ocp_version,
+    skipif_managed_service,
+    skipif_disconnected_cluster,
+    skipif_proxy_cluster,
+)
+
+from ocs_ci.ocs.resources import pod
+from ocs_ci.helpers.helpers import (
+    create_pods,
+)
+
+log = logging.getLogger(__name__)
+
+
+@tier1
+@skipif_ocs_version("<4.11")
+@skipif_ocp_version("<4.11")
+@skipif_managed_service
+@skipif_disconnected_cluster
+@skipif_proxy_cluster
+class TestNfsEnable(ManageTest):
+    """
+    Test nfs feature enable for ODF 4.11
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup(
+        self,
+    ):
+        """ """
+        self.namespace = "openshift-storage"
+        self.storage_cluster_obj = ocp.OCP(
+            kind="Storagecluster", namespace=self.namespace
+        )
+        self.rook_ceph_cm_obj = ocp.OCP(kind="Configmap", namespace=self.namespace)
+        self.nfs_plugin_pod_obj = ocp.OCP(kind="Pod", namespace=self.namespace)
+        self.nfs_plugin_provisioner_pod_obj = ocp.OCP(
+            kind="Pod", namespace=self.namespace
+        )
+        self.nfs_ganesha_pod_obj = ocp.OCP(kind=constants.POD, namespace=self.namespace)
+
+        self.nfs_sc = "ocs-storagecluster-ceph-nfs"
+
+        self.nfs_spec_enable = '{"spec": {"nfs":{"enable": true}}}'
+        self.rook_csi_config_enable = '{"data":{"ROOK_CSI_ENABLE_NFS": "true"}}'
+        self.nfs_spec_disable = '{"spec": {"nfs":{"enable": false}}}'
+        self.rook_csi_config_disable = '{"data":{"ROOK_CSI_ENABLE_NFS": "false"}}'
+
+        pod_functions = {
+            "cephfsplugin": partial(
+                pod.get_plugin_pods, interface=constants.CEPHBLOCKPOOL
+            ),
+            "cephfsplugin_provisioner": partial(pod.get_cephfsplugin_provisioner_pods),
+        }
+
+        # Get number of pods of type 'cephfs plugin' and 'cephfs plugin provisioner' pods
+        self.num_of_cephfsplugin_pods = len(pod_functions["cephfsplugin"]())
```

Comment:
why `num_of_cephfsplugin_pods`? I guess that should that be `num_of_nfsplugin_pods`, and similar in the rest of the file...

Reply:
Number of nfs_plugin pods and nfs_provisonerplugin pods are same as number of cephfsplugin pods and cephfs_provisionerplugin pods respectively for a cluster.
Have fetched the number of cephfs plugin pods and cephfs_provisionerplugin pods to assert same numbers of nfs plugin pods and nfs_provisonerplugin pods are up and running.

Reply:
That is more of a coincidence than anything else. There is no requirement that cephfs provisioners are available at all for NFS. This probably works fine, but I think it would be better/cleaner to check on the number of NFS pods.

Reply:
Addressed in latest commit.

Reply:
thanks!